diff --git a/node_modules/semantic-release/lib/git.js b/node_modules/semantic-release/lib/git.js
index fa25173..2e5f3b9 100644
--- a/node_modules/semantic-release/lib/git.js
+++ b/node_modules/semantic-release/lib/git.js
@@ -223,7 +223,7 @@ export async function verifyAuth(repositoryUrl, branch, execaOptions) {
  * @throws {Error} if the tag creation failed.
  */
 export async function tag(tagName, ref, execaOptions) {
-  await execa("git", ["tag", tagName, ref], execaOptions);
+  await execa("git", ["tag", "-a", tagName, ref, "-m", `release ${tagName}`], execaOptions);
 }
 
 /**
@@ -308,7 +308,8 @@ export async function isBranchUpToDate(repositoryUrl, branch, execaOptions) {
  */
 export async function getNote(ref, execaOptions) {
   try {
-    return JSON.parse((await execa("git", ["notes", "--ref", GIT_NOTE_REF, "show", ref], execaOptions)).stdout);
+    const tagHead = await getTagHead(ref, execaOptions);
+    return JSON.parse((await execa("git", ["notes", "--ref", GIT_NOTE_REF, "show", tagHead], execaOptions)).stdout);
   } catch (error) {
     if (error.exitCode === 1) {
       return {};
